addons/gamification/models/res_users.py

```diff
@@ -12,13 +12,45 @@ class Users(models.Model):
     gold_badge = fields.Integer('Gold badges count', compute="_get_user_badge_level")
     silver_badge = fields.Integer('Silver badges count', compute="_get_user_badge_level")
     bronze_badge = fields.Integer('Bronze badges count', compute="_get_user_badge_level")
+    rank_id = fields.Many2one('gamification.karma.rank', 'Rank')
+    next_rank_id = fields.Many2one('gamification.karma.rank', 'Next Rank')
+
+    @api.model_create_multi
+    def create(self, values_list):
+        res = super(Users, self).create(values_list)
+        res._recompute_rank()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        if 'karma' in vals:
+            self._recompute_rank()
+        return super(Users, self).write(vals)
+
+    def _rank_changed(self):
+        """Hook for other modules
+            e.g. Website_slides to send 'new rank reached' email
+        """
+        pass
 
     @api.multi
     def add_karma(self, karma):
         for user in self:
             user.karma += karma
         return True
 
+    def _recompute_rank(self):
```

Comment:
I would remove users having 0 karma to be sure. Probably with a search to avoid a filtered / mapped.